crypto/ec/ecdsa_ossl.c

```diff
@@ -126,22 +127,33 @@ static int ecdsa_sign_setup(EC_KEY *eckey, BN_CTX *ctx_in,
         goto err;
 
     do {
```

Comment:
I have to believe there is a simpler flow of control that doesn't require nested do/while loops.

Reply:
It is unlikely that `BN_is_zero(k) is true` as k is the BN converted from hmac result.
So it won't loop generally.

Reply:
I am not talking about the number of repetitions, I am talking about the control flow as written.

Reply:
@mspncp @richsalz 
I re-read the code and finally realized, the K won't be zero, so it isn't necessary to loop here.
But I found out another problem:
if the generated K results in r=0 (r of the `(r,s)` pair), it will call bn_generate_dsa_deterministic_nonce again and get the exactly same K, thus fail to jump out of the loop!

Reply:
@richsalz the nested 'do' loops date way back to the initial revision 4d94ae00d56 of this file. While I agree with your opinion, I think a cleanup would be a task for a separate pull request.

https://github.com/openssl/openssl/blob/4d94ae00d5614d64d4dd065860c4b00161a81f82/crypto/ecdsa/ecs_ossl.c#L173-L194

Reply:
@mspncpI I found my opinion is incorrect xD
My `bn_generate_dsa_deterministic_nonce` won't generate `K=0`, 
but the original `BN_generate_dsa_nonce` may. 
I wonder why don't `BN_generate_dsa_nonce` generate `0<K<range` but `0<=K<range`
it seems very awful